sklearn/preprocessing/_encoders.py

```diff
@@ -83,11 +110,18 @@ def _fit(self, X, handle_unknown='error', force_all_finite=True):
                                  " it has to be of shape (n_features,).")
 
         self.categories_ = []
+        category_counts = []
```

Comment:
@thomasjpfan would it be desirable to make `category_counts` an attribute? this would help with implementation of #18679 `most_frequent` drop method.

Reply:
We can convert this to an attribute in a PR that requires it such as #18679.